gammapy/cube/fit.py

```diff
@@ -740,14 +740,19 @@ def read(cls, filename, name=None):
     def from_dict(cls, data, components, models):
         """Create from dicts and models list generated from YAML serialization."""
         dataset = cls.read(data["filename"], name=data["name"])
-        bkg_name = data["background"]
-        model_names = data["models"]
-        models_list = [model for model in models if model.name in model_names]
+        models_list = [
```

Comment:
I think this logic of selecting the model component should rather be moved to the the models setter of each dataset (search for `@models.setter`). This ensures a consistent behaviour between all datasets.